torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -92,150 +97,201 @@ void PropagateShapeOnNode(Node * node) {
     return setDynamicType(node);
   }
 
+  bool handled = false;
```

Comment:
Yes, but in most cases you'll just use `check_overload` and have it all handled for you, while otherwise you can easily forget a `return` because it's going to be needed at the end of *every if branch* (otherwise you *want to fall through*), and you'll have the auto-inference run anyway. I can change it if you want it, but this it feels less error prone this way

Reply:
I don't have strong opinions, we can keep it as is.